src/appleseed/renderer/kernel/intersection/tracecontext.h

```diff
@@ -54,7 +57,7 @@ class APPLESEED_DLLSYMBOL TraceContext
 {
   public:
     // Constructor, initializes the trace context for a given scene.
-    explicit TraceContext(const Scene& scene);
+    explicit TraceContext(const Project& project);
```

Comment:
Why this change ? If I understand correctly, we only need the scene for `get_scene`.

Reply:
We need `Project` here to pass it down to `AssemblyTree` -- all this to, as you've guessed in one of the comments, to provide `TriangleTree` with the `TextureStore`.

> If I understand correctly, we only need the scene for get_scene.
I'm not sure what do you mean by this.